viewing_party/party.py

```diff
@@ -1,23 +1,121 @@
-# ------------- WAVE 1 --------------------
+# ---------------------------------------------------- WAVE 1 ----------------------------------------------------
+#-----------------------------------------------------------------------------------------------------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    
+    new_movie_info = {"title": title, "genre": genre, "rating": rating}
+    return new_movie_info
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
 
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)

```

Comment:
So now that we have talked about Big O analysis, we know that `.remove` has a time complexity of `O(n)` since it needs to look through every element in a list to find the sought after value. Right now, this loop has an outer loop of  `O(n)` and you when you find the movie to remove you have another `O(n)` operation. How could we circumvent using `remove`? (Hint: Think about list comprehension) 

Also in this implementation, we will only use `remove` once, but in other filtering problems we might have to do this for multiple elements so it doesn't hurt to get in the habit now.

Reply:
question: will ask about it in office hours

Reply:
fixed this thanks for explaining!